### PR TITLE
[RTM] Release/2.0.0

### PR DESCRIPTION
**Pull request by discordier**
This PR is the work in progress for the new version 2.0 which will provide compatibility with Contao 3 & 4.

Note that several things have gotten removed and needs to get deprecated in the current version:
- The `CreateEventDispatcherEvent` is no more (as discussed in #7).
- The `event-dispatcher.initializer` service has been removed (not documented anyway).
- The `event-dispatcher.factory.default` service has been removed (not documented anyway).
- The `event-dispatcher.configurator.default` service has been removed (not documented anyway).
- The `event-dispatcher.factory` service has been removed (not documented anyway).
- The `event-dispatcher.configurator` service has been removed (not documented anyway).

The services which got removed were a nice idea for maximum flexibility when not having a container compiler but got in the way as we now have such a compiler in symfony, so I had to remove them.

### CODE REVIEW

**Comment by baumannsven** [comment]
@discordier What do you mean for this?

**Comment by zonky2** [comment]
How can I test or check it?

File: src/Resources/contao/config/services.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+use ContaoCommunityAlliance\Contao\EventDispatcher\Configuration\ResourceLocator;
+use ContaoCommunityAlliance\Contao\EventDispatcher\EventDispatcherPopulator;
+use DependencyInjection\Container\PimpleGate;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+
+/** @var PimpleGate $container */
+
+// Contao 4 code.
+if ($container->isContao4()) {
```
Comment (dmolineus) [inline src/Resources/contao/config/services.php]: This if statement isn't necessary as it's only for Contao 4 (composer.json).

File: src/ContaoManager/Plugin.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+     * {@inheritdoc}
+     */
+    public function getBundles(ParserInterface $parser)
+    {
+        return [
+            BundleConfig::create(CcaEventDispatcherBundle::class)
+                ->setLoadAfter(
+                    [
+                        ContaoCoreBundle::class,
+                        ContaoManagerBundle::class,
```
Comment (dmolineus) [inline src/ContaoManager/Plugin.php]: Is it required to set the manager bundle in the setLoadAfter section in general or is there a specific reason? I just don't know the details how the manager bundle works.

File: composer.json [hunk 3]
```diff
@@ -40,7 +40,7 @@
         "contao/manager-bundle": "~4.4"
     },
     "replace": {
-        "bit3/contao-event-dispatcher": "self.version"
+        "bit3/contao-event-dispatcher": "1.0.x-dev"
```
Comment (baumannsven) [inline composer.json]: @discordier If replace as self.version, the package not install as symlink. What do you mean?

**Comment by baumannsven** [comment]
#7

File: src/ContaoManager/Plugin.php (hunk 2 above)
Comment (baumannsven) [inline src/ContaoManager/Plugin.php]: This plugin loads if installed contao managed edition.
The managed edition show for [configuration](https://github.com/contao-community-alliance/event-dispatcher/blob/feature/contao4/composer.json#L56) in composer.json.

File: src/Resources/contao/config/services.php (hunk 1 above)
Comment (discordier) [inline src/Resources/contao/config/services.php]: Has been removed in 2e67095

**Comment by discordier** [comment]
@dmolineus Could you update your review?

File: composer.json [hunk 4]
```diff
@@ -5,55 +5,54 @@
...
     },
     "require": {
-        "php": ">=5.3",
-        "contao/core": ">=2.11,<4-dev",
-        "contao-community-alliance/composer-plugin": "~2.0",
-        "contao-community-alliance/dependency-container": "~1.6",
-        "symfony/event-dispatcher": "~2.3"
+        "php": ">=5.6",
+        "contao/core-bundle": "~4.4",
+        "contao-community-alliance/dependency-container": "~2.0@dev",
```
Comment (dmolineus) [inline composer.json]: Is it required? Think we could dropt the requirement.

File: src/Resources/config/services.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+services:
+    contao_community_alliance.event_dispatcher.populator:
```
Comment (dmolineus) [inline src/Resources/config/services.yml]: Maybe use cca as shortcut here. As I said would be nice to discuss our naming conventions.Question  ;-)

Comment (discordier) [inline src/Resources/config/services.yml]: We already have the shortcut almost everywhere, this one however slipped through :smile_cat:

Comment (discordier) [inline src/Resources/config/services.yml]: Adressed in b6ca969

File: composer.json (hunk 4 above)
Comment (discordier) [inline composer.json]: It is used by the `src/Resources/contao/services.php` which starts the whole thing up.

Comment (discordier) [inline composer.json]: Added a test to show it in: 90f6213

**Comment by discordier** [comment]
:100: % code coverage :tada:

**Comment by discordier** [comment]
If flagged this now RTM.

If the missing ppl would please either approve or request changes?
@baumannsven @stefanheimes @zonky2

**Comment by zonky2** [comment]
how can I test it to get a review?

**Comment by discordier** [comment]
Could the remaining parties @baumannsven and @stefanheimes please make their final review so we can release this?

**Pull request merged**